pkg/vault/create/create.go

```diff
@@ -117,11 +116,23 @@ func (p *VaultCreationParam) validate() error {
 		}
 	}
 
+	if p.KubeProvider == cloud.AWS {
+		if p.AWS == nil {
+			validationErrors = append(validationErrors, errors.Errorf("%s selected as kube provider, but no %s specific parameters provided", cloud.AWS, cloud.AWS))
+		}
+		if err := p.AWS.validate(); err != nil {
+			validationErrors = append(validationErrors, err)
+		}
+	}
+
 	return util.CombineErrors(validationErrors...)
 }
 
 func (p *GKEParam) validate() error {
 	var validationErrors []error
+	if p == nil {
```

Comment:
I don't think this is technically possible. _p_ is the instance pointer on which the _validate_ method is called on. If this instance pointer is _nil_, you are getting a panic directly, not in this function.

That said, we could add some unit tests for `GKEParam` and `AWSParam`. WDYT?

Reply:
In Go is possible to call a method on a null pointer receiver. It is perfectly allowed. Actually sometimes is a good practice to check the receiver for nil before doing anything into that method. https://medium.com/@mehdy/the-story-of-a-nil-pointer-receiver-b530e00a5f11

Reply:
As @ccojocar mentioned the method on the nil pointer receiver can be executed. I will quote Dave Cheney 

> Methods in Go are just syntactic sugar for a function which passes the receiver as the first formal parameter.

Reference: https://dave.cheney.net/tag/receivers

The brief comment from Francesc Campoy Flores https://youtu.be/ynoY2xz-F8s?t=987 watch till 18:36 :) 

Regarding the unit tests, I could write them, but there is already `create_test.go` file which is an integration test with GKE. I could move `create_test.go` to `create_intergration_test.go`, so that `create_test.go` will have new unit tests. 

I'm not sure if CloudBees could have integration tests with AWS environment?

WDYT?  @ccojocar @hferentschik

Reply:
Look at that. Thanks for filling this gap of mine.